app/src/main/java/com/mediapipe/example/sign_language/SignLanguageHelper.kt

```diff
@@ -0,0 +1,102 @@
+package com.mediapipe.example.sign_language
+
+import android.content.Context
+import com.mediapipe.example.sign_language.LandmarkIndex.LEFT_HANDLANDMARK_INDEX
+import com.mediapipe.example.sign_language.LandmarkIndex.RIGHT_HANDLANDMARK_INDEX
+import org.tensorflow.lite.Interpreter
+import org.tensorflow.lite.support.common.FileUtil
+import java.io.FileInputStream
+import java.nio.MappedByteBuffer
+import java.nio.channels.FileChannel
+import kotlin.math.max
+import kotlin.math.min
+
+class SignLanguageHelper(private val context: Context) {
+    private lateinit var interpreter: Interpreter
+    private lateinit var associatedAxisLabels: List<String>
+    private lateinit var listener: SignLanguageListener
+    fun createInterpreter(listener: SignLanguageListener) {
+        interpreter = Interpreter(
+            loadModelFile(context),
+            Interpreter.Options()
+        )
+
+        associatedAxisLabels =
+            FileUtil.loadLabels(context, "sign_language_labels.txt")
+
+        this.listener = listener
+    }
+
+    fun runInterpreter(landmarks: MutableList<Array<FloatArray>>) {
+        // trim the input
+        if (landmarks.isEmpty()) return
+        val minFrame = 10
+        var firstFrame = -1
+        var lastFrame = -1
+
+        // check first and last occurrences of hand landmarks
+        landmarks.forEachIndexed { index, floatArrays ->
+            if (!floatArrays[LEFT_HANDLANDMARK_INDEX][0].isNaN() || !floatArrays[RIGHT_HANDLANDMARK_INDEX][0].isNaN()) {
```

Comment:
Looking at this, do we even need to use the holistics API, or would hand alone work? Some sign language gestures use position of the hand on the face, but I don't see those face landmarks being used here.